src/backends/mysql/blob.cpp

```diff
@@ -18,43 +21,99 @@
 using namespace soci;
 using namespace soci::details;
 
-mysql_blob_backend::mysql_blob_backend(mysql_session_backend &session)
-    : session_(session)
+mysql_blob_backend::mysql_blob_backend(mysql_session_backend &)
+    : details::trivial_blob_backend()
 {
-    throw soci_error("BLOBs are not supported.");
 }
 
 mysql_blob_backend::~mysql_blob_backend()
 {
 }
 
-std::size_t mysql_blob_backend::get_len()
+static unsigned char decode_hex_digit(char c)
 {
-    throw soci_error("BLOBs are not supported.");
+    unsigned char i = static_cast<unsigned char>(tolower(c));
+
+    assert((i >= '0' && i <= '9') || (i >= 'a' && i <= 'f'));
+
+    if (i <= '9') {
+        return i - '0';
+    } else {
+        return i - 'a';
+    }
 }
 
-std::size_t mysql_blob_backend::read(
-    std::size_t /* offset */, char * /* buf */, std::size_t /* toRead */)
+static char encode_hex_digit(unsigned char d) {
+    static const char hexMap[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
```

Comment:
It takes one more byte, but I think
```suggestion
    static const char hexMap[] = "0123456789abcdef";
```
is quite a bit more readable.